crates/model/src/order.rs

```diff
@@ -667,12 +667,15 @@ impl OrderClass {
     }
 }
 
+#[serde_as]
 #[derive(Eq, PartialEq, Clone, Debug, Default, Hash, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct LimitOrderClass {
     #[serde(with = "u256_decimal")]
     pub surplus_fee: U256,
```

Comment:
From an `GET api/v1/orders/$orderUid` API perspective, I think it would be cleaner if `surplus_fee` would be the `executed_surplus_fee` if it is available and the `surplus_fee` otherwise. I think technically there is a possible race condition where we may return the wrong fee for the explorer, but that isn't critical and not worth the extra field IMO.

From the `GET api/v1/auction` perspective, we should always use the `surplus_fee` since this is the "up-to-date" one for auctions.

This would require #529 though.

Reply:
On the other hand both might be relevant for the frontend. Maybe they also want to show a user what the currently required surplus to settle would be. Agree we can and should make this nicer when we have separate structs. I do  like the name `executed_` either way because it fits the other fields that are populated after execution.

Reply:
Actually, thinking a bit more about this, we likely need a separate `executed_surplus_fee` field just because once we introduce partially fillable orders, we would need to track the total amount of executed surplus fees, and it makes sense to use one field.